compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -38,6 +39,10 @@ pub struct Interpreter<'interner> {
     /// up all currently visible definitions.
     scopes: &'interner mut Vec<HashMap<DefinitionId, Value>>,
 
+    /// The scope of --debug-comptime, or None if unset
+    pub(super) debug_comptime_scope: Option<FileId>,
+    pub(super) debug_comptime_evaluations: &'interner mut Vec<InterpreterError>,
+
```

Comment:
Is this needed? It looks like the interpreter itself never actually uses these fields. I'd imagine only the scan pass or the elaborator would check after the interpreter has ran.

Reply:
They are needed in the "middle" of the comptime scan, so would end up as parameters to many of the scan functions. If that's preferable, I'll move them there.

Reply:
My question is if this is only needed for scanning, why is it in the interpreter?
Looking at `interpreter.rs`, these fields are unused within that file.

Reply:
Because scanning is defined as interpreter `impl`'s.

Reply:
Just pulled the PR locally and realized this. I think it can stay for now but we should remove it when we remove the scanning pass since we don't really support macro stuff in legacy code anyway.